### PR TITLE
Add an option to always apply splitting to HTTPS

### PR DESCRIPTION
**Pull request by bemasc**

### CODE REVIEW

File: android/intra.go [hunk 1]
```diff
@@ -35,12 +35,12 @@ type IntraTunnel interface {
 //
 // Throws an exception if the TUN file descriptor cannot be opened, or if the tunnel fails to
 // connect.
-func ConnectIntraTunnel(fd int, fakedns, udpdns, tcpdns string, listener tunnel.IntraListener) (IntraTunnel, error) {
+func ConnectIntraTunnel(fd int, fakedns, udpdns, tcpdns string, alwaysSplitHTTPS bool, listener tunnel.IntraListener) (IntraTunnel, error) {
```
Comment (alalamav) [inline android/intra.go]: nit: should the name be `alwaysSplitHttps`?

File: tunnel/intra/direct_split.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	return s.conn.Read(buf)
+}
+
+func (s *splitter) CloseRead() error {
+	return s.conn.CloseRead()
+}
+
+// Write-related functions
+func (s *splitter) Write(b []byte) (int, error) {
+	if s.fresh {
```
Comment (alalamav) [inline tunnel/intra/direct_split.go]: I think flipping the condition for an early return would make this more readable.

File: tunnel/intra/direct_split.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+import (
+	"io"
+	"net"
+)
+
+type splitter struct {
+	DuplexConn
+	conn  *net.TCPConn
+	fresh bool
```
Comment (alalamav) [inline tunnel/intra/direct_split.go]: This variable tracks whether we have read from/written to this connection, right? Consider renaming or adding a comment.

File: android/intra.go (hunk 1 above)
Comment (bemasc) [reply android/intra.go]: That's what I had originally, but `go lint` demanded this version.

File: tunnel/intra/direct_split.go (hunk 2 above)
Comment (bemasc) [reply tunnel/intra/direct_split.go]: Done

File: tunnel/intra/direct_split.go (hunk 3 above)
Comment (bemasc) [reply tunnel/intra/direct_split.go]: Done.  (Also this was initialized to false!)

**Pull request merged**